packages/pwa/app/utils/url.js

```diff
@@ -102,16 +103,23 @@ export const productUrlBuilder = (product) => encodeURI(`/product/${product.id}`
 export const searchUrlBuilder = (searchTerm) => `/search?q=${searchTerm}`
 
 /**
- * Returns a URL using the new locale.
+ * Returns a URL pathname using the new locale.
  *
- * @param previousLocale
- * @param newLocale
+ * @param {string} newLocale
+ * @param {Object} [options]
+ * @param {Object} options.location - original location (similar value to `window.location`)
  * @returns {string}
  */
-export const buildUrlLocale = (previousLocale, newLocale) => {
+export const buildUrlLocale = (newLocale, options = {}) => {
```

Comment:
FYI it turns out that this change overlaps with Adam's changes in his PR: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/105#discussion_r717789392

I'm thinking he can extract his changes to this function and have it in a separate PR. Once that PR is merged, I'll merge it in to my PR and make other changes as necessary.

Reply:
Hey @vmarta The changes look good to me. Let me know if you want me to take another look after you sort things out with another PR.